src/daemon/bitcoind/poller.rs

```diff
@@ -386,23 +423,92 @@ fn mark_confirmed_emers(
     Ok(())
 }
 
+fn maybe_cpfp_txs(
+    revaultd: &Arc<RwLock<RevaultD>>,
+    bitcoind: &BitcoinD,
+    needs_cpfp_cache: &mut HashMap<Txid, CpfpTxInfo>,
+) -> Result<(), BitcoindError> {
+    let revaultd = revaultd.read().unwrap();
+    log::debug!("Checking if transactions need CPFP...");
+
+    if revaultd.cpfp_key.is_none() {
+        log::error!("We should CPFP transactions, but we don't have a cpfp key!");
+        // Life is not really ok, but we don't want to crash either...
+        return Ok(());
+    }
+
+    let current_blockheight = bitcoind.get_tip()?.height;
+
+    for (txid, info) in needs_cpfp_cache.iter_mut() {
+        if info.feebumped_blockheight.is_none()
+            && info.broadcasted_blockheight + 1 < current_blockheight
+        {
+            log::info!("CPFPing transaction with id '{}'", txid);
+            let feerate = bitcoind.estimate_feerate()?;
+            let listunspent: Vec<_> = bitcoind
+                .listunspent(None)?
+                .into_iter()
+                .filter(|u| u.label == CPFP_UTXOS_LABEL)
+                .collect();
+            use crate::daemon::control::RpcControlError;
+            let tx = match get_cpfp_transaction(txid, &revaultd, feerate, listunspent) {
+                Ok(tx) => tx,
+                Err(RpcControlError::InsufficientFunds) => {
+                    // Well, we're poor.
+                    log::error!(
+                        "We wanted to feebump transaction '{}', but we don't have enough funds!",
+                        txid
+                    );
+                    return Ok(());
+                }
+                Err(RpcControlError::MissingCpfpInput) => {
+                    // We thought we never CPFPed this transaction (feebumped_blockheight is None!), but actually,
+                    // *shit happens*, and the cpfp outpoint has already been spent. Could be:
+                    // - If a transaction is dropped from the mempool we're going to re-insert
+                    // it in the cache with "feebumped_blockheight: None", even tho maybe it already has
+                    // a feebump tx
+                    // - In a really high fee scenario, we might have used this transaction's feebump
+                    // output for feebumping another transaction, effectively cpfping this one as well
+                    log::warn!("We wanted to feebump transaction '{}', but its CPFP output is not in listunspent!", txid);
```

Comment:
You could avoid these edge cases most of the case by calling `listunspent` with `minconf = 0`. You still need to handle race conditions though :/

Reply:
I am calling listunspent with minconf = 0 already (otherwise I would never find the outputs I need to use to feebump, as I feebump only unconf txs 😄 ). The second scenario is not something I can really prevent... I accidentally feebumped this transaction while feebumping another one. It's fine. Actually, it's great 😄